src/pages/organisationUnits/list/OrganisationUnitList.tsx

```diff
@@ -167,35 +140,45 @@ export const OrganisationUnitList = () => {
 
     const computedRoot = useMemo(() => {
         return flatOrgUnits.filter(
-            (ou) => !ou.parent || userRootOrgUnitIds.includes(ou.id)
+            (ou) => !ou.parent || initialExpandedState[ou.id]
         )
-    }, [flatOrgUnits, userRootOrgUnitIds])
+    }, [flatOrgUnits, initialExpandedState])
 
-    // handle when expanded
     const handleExpand = useCallback(
         (valueOrUpdater: Updater<ExpandedState>) => {
-            // we are just using ExpandedStateList in state, because we need the exact Ids to load
-            // but API exposes ExpandedState (includes true to expand all)
-            // so we handle and translate that to expand all loaded units
-            const setExpandedFunc = isFiltering
-                ? setExpandedDuringFilter
-                : setExpanded
-            const expandAll = () =>
-                Object.fromEntries(
-                    flatOrgUnits.map((ou) => [ou.id, true] as const)
-                )
+            // when we expand something and are not filtering, we need to load the children
+            // also translate expandedState === true (expand all) to expand all loaded units
+            const getAllExpanded = () =>
+                Object.fromEntries(flatOrgUnits.map((ou) => [ou.id, true]))
+            // we always want to keep root loaded
+            const getValueWithRoot = (value: ExpandedStateList) => ({
+                ...initialExpandedState,
+                ...value,
+            })
             if (typeof valueOrUpdater === 'function') {
-                setExpandedFunc((old) => {
+                setExpanded((old) => {
                     const value = valueOrUpdater(old)
-                    return value === true ? expandAll() : value
+                    if (!isFiltering) {
+                        setParentIdsToLoad(
```

Comment:
It seems a bit weird to call a state-updater within a stateUpdate (`setExpanded`). But if the `valueOrUpdater` is a function, we either have to do this, or put `expanded` as a dependency and call the `valueOrUpdater(expanded)` I think?